articles/connecting-a-self-hosted-gitlab-repository-as-your-source.mdx

```diff
@@ -51,19 +51,45 @@ To connect a self-hosted GitLab repository and start syncing files, follow these
 2. Select **Self Hosted GitLab repository**
 3. Click **Authenticate**
 
-<comp.DocsImage path="ye_olde_images/self-hosted-gitlab.png" alt="Selecting self hosted GitLab as the your provider" type="screenshot" />
+<comp.DocsImage path="ye_olde_images/gitlab-self-hosted-connect.png" alt="Settings page from Gitlab" type="screenshot" />
 
-This redirects you to your self-hosted GitLab repository. Log in and authorize CloudCannon access to your self-hosted GitLab account. You'll be redirected back to CloudCannon to pick a repository to connect. If you don't have one for this website, create a new one in your self-hosted GitLab and refresh this page.
 
-<comp.DocsImage path="ye_olde_images/self-hosted-gitlab-selected.png" alt="Selecting self hosted GitLab repository to sync" type="screenshot" />
+This will open a menu where you can connect to GitLab, via one of two methods.
+
+### Connecting with your user account
+
+You can connect to GitLab by authenticating with your user account. This will give your CloudCannon organization access to the same GitLab repositories you have access to.
+
+This redirects you to your self-hosted GitLab repository. Log in and authorize CloudCannon access to your self-hosted GitLab account. You'll be redirected back to CloudCannon to pick a repository to connect.
+
+<comp.DocsImage path="ye_olde_images/files-source-syncing-gitlab-oauth.png" alt="A menu with a button labelled 'Authenticate with GitLab'." type="screenshot" />
+
+### Authenticating with a group access token
+
+Alternatively, you can create a group access token in GitLab and save that token in CloudCannon. This will give your CloudCannon organization access to all the resources scoped to your token.
```

Comment:
```suggestion
Alternatively, you can create a group access token in GitLab and save that token in CloudCannon. This will give your CloudCannon organization access to all the resources scoped to your token.
See GitLab's [Group access tokens](https://docs.gitlab.com/ee/user/group/settings/group_access_tokens.html) documentation for how to create and configure a token.
<comp.Notice info_type="important">
  For full functionality on CloudCannon, your group access token should have (at least) the Maintainer role, and the `api`, `read_api`, `read_repository`, and `write_repository` scopes.
</comp.Notice>
```